packages/angular/cli/commands/doc-impl.ts

```diff
@@ -23,10 +23,16 @@ export class DocCommand extends Command<DocCommandSchema> {
     let domain = 'angular.io';
 
     if (options.version) {
+      // version can either be a string containing "next"
       if (options.version == 'next') {
         domain = 'next.angular.io';
-      } else {
+      // or a number where version must be a valid Angular version (i.e. not 0, 1 or 3)
+      } else if (!isNaN(+options.version) && ![0, 1, 3].includes(+options.version)) {
```

Comment:
Shouldn't need to revalidate.

Reply:
Sadly I think we need to, because the CLI support for JSON schema validation is really minimal. It was possible to do `ng doc --version whatever` or `ng doc --version 3` despite the schema. I talked about that with @clydin and we came to the conclusion that we needed to validate in code as well.